tests/Unit/Evolution/DiscontinuousGalerkin/Test_ConservationHelper.cpp

```diff
@@ -0,0 +1,166 @@
+                 ::Tags::NormalDotFlux<Tags::Var2<Dim>>, AbsCharSpeed>;
+  using dg_package_data_temporary_tags = tmpl::list<>;
+  using dg_package_data_volume_tags = tmpl::list<Tags::VolumeDouble>;
+
+  double dg_package_data(
+      const gsl::not_null<Scalar<DataVector>*> packaged_var1,
+      const gsl::not_null<Scalar<DataVector>*> packaged_normal_dot_flux_var1,
+      const gsl::not_null<tnsr::i<DataVector, Dim, Frame::Inertial>*>
+          packaged_var2,
+      const gsl::not_null<tnsr::i<DataVector, Dim, Frame::Inertial>*>
+          packaged_normal_dot_flux_var2,
+      const gsl::not_null<Scalar<DataVector>*> packaged_abs_char_speed,
+      const Scalar<DataVector>& var1,
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& var2,
+      const tnsr::I<DataVector, Dim, Frame::Inertial>& flux_var1,
+      const tnsr::Ij<DataVector, Dim, Frame::Inertial>& flux_var2,
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& normal_covector,
+      const boost::optional<tnsr::I<DataVector, Dim, Frame::Inertial>>&
+      /*mesh_velocity*/,
+      const boost::optional<Scalar<DataVector>>& normal_dot_mesh_velocity,
+      const double volume_double) const noexcept {
+    *packaged_var1 = var1;
+    *packaged_var2 = var2;
+    dot_product(packaged_normal_dot_flux_var1, flux_var1, normal_covector);
+    for (size_t i = 0; i < Dim; ++i) {
+      packaged_normal_dot_flux_var2->get(i) =
+          flux_var2.get(i, 0) * get<0>(normal_covector);
+      for (size_t j = 1; j < Dim; ++j) {
+        packaged_normal_dot_flux_var2->get(i) +=
+            flux_var2.get(i, j) * normal_covector.get(j);
+      }
+    }
+
+    if (static_cast<bool>(normal_dot_mesh_velocity)) {
+      get(*packaged_abs_char_speed) =
+          abs(volume_double * get(var1) - get(*normal_dot_mesh_velocity));
+    } else {
+      get(*packaged_abs_char_speed) = abs(volume_double * get(var1));
+    }
+    return max(get(*packaged_abs_char_speed));
+  }
+
+  void dg_boundary_terms(
+      const gsl::not_null<Scalar<DataVector>*> boundary_correction_var1,
+      const gsl::not_null<tnsr::i<DataVector, Dim, Frame::Inertial>*>
+          boundary_correction_var2,
+      const Scalar<DataVector>& var1_int,
+      const Scalar<DataVector>& normal_dot_flux_var1_int,
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& var2_int,
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& normal_dot_flux_var2_int,
+      const Scalar<DataVector>& abs_char_speed_int,
+      const Scalar<DataVector>& var1_ext,
+      const Scalar<DataVector>& normal_dot_flux_var1_ext,
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& var2_ext,
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& normal_dot_flux_var2_ext,
+      const Scalar<DataVector>& abs_char_speed_ext,
+      const dg::Formulation dg_formulation) const noexcept {
+    if (dg_formulation == dg::Formulation::WeakInertial) {
+      get(*boundary_correction_var1) =
```

Comment:
A short comment explaining this math might be helpful. Are you coding up a particular flux correction? Is this an arbitrary algebraic expression with the correct symmetries (if so, which?)? I'm thinking of helping to guide someone who might need to edit this code in the future.